Realm/RLMArray.mm

```diff
@@ -299,6 +299,13 @@ - (RLMResults *)objectsWhere:(NSString *)predicateFormat args:(va_list)args
     return [self objectsWithPredicate:[NSPredicate predicateWithFormat:predicateFormat arguments:args]];
 }
 
+- (id)valueForKeyPath:(NSString *)keyPath {
+    if (_backingArray) {
+        return [_backingArray valueForKeyPath:keyPath];
```

Comment:
@jpsim: Only `@count` works with `valueForKey:`. So not supporting nested key paths for the sake of consistency for standalone arrays wouldn't be a breaking semantic change.